dash-pipeline/SAI/sai_api_gen.py

```diff
@@ -27,6 +27,35 @@
 PARAM_ACTIONS = 'paramActions'
 OBJECT_NAME_TAG = 'objectName'
 SCOPE_TAG = 'scope'
+STRUCTURED_ANNOTATIONS_TAG = 'structuredAnnotations'
+KV_PAIRS = 'kvPairs'
+KV_PAIR_LIST = 'kvPairList'
+
+sai_type_to_field = {
+    'bool': 'booldata',
+    'sai_uint8_t': 'u8',
+    'sai_object_id_t': 'u16',
+    'sai_uint16_t': 'u16',
+    'sai_ip_address_t': 'ipaddr',
+    'sai_ip_addr_family_t': 'u32',
+    'sai_uint32_t': 'u32',
+    'sai_mac_t': 'mac'
+}
+
+def p4_annotation_to_sai_attr(p4rt, sai_attr):
+    for anno in p4rt[STRUCTURED_ANNOTATIONS_TAG]:
+        if anno[NAME_TAG] == 'sai':
```

Comment:
Suggest using a constant e.g. SAI_ANNOTATION. Also please use @Sai not @sai. Custom annotations should start with upper-case per P4 language spec, see https://p4.org/p4-spec/docs/P4-16-v1.2.1.html#sec-predefined-annotations

Reply:
Done